weblab-docker/Makefile.inc

```diff
@@ -56,6 +56,17 @@ build::
 	  -t $(IMAGE_NAME):$(IMAGE_TAG) $(BUILD_PATH)
 	docker $(DOCKER_ARGS) image tag $(IMAGE_NAME):$(IMAGE_TAG) $(IMAGE_NAME):latest
 
+# Builds the Docker image 
+build-main-branch-github::
```

Comment:
I tried to figure out a way to make it work with environment variables but it doesn't seem like Docker supports such use cases. We want to use `COPY` in one of the branches and `RUN` in the other branch. The approach with "if" and "else" only seems to work if we have two `RUN` commands.

I guess I could make it always copy the parent directory and then add a `RUN` afterwards which deletes the just-copied folder and clones the repo in its place if an environment variable is set, but this feels really hacky and I'm not sure if it's better than our current approach. What do you think?

Reply:
Mauricio says:
> Copying files and restoring them after seems like the hackiest thing we could do.
> 
> I’d say let’s always clone the repo then. We don’t build this image that often anyways. CI may be a couple seconds slower (or not, for GitHub, clone is a local call probably) but better than a workaround.

I changed it to clone the repo and to have an environment variable to checkout a particular branch (we need this for the CI script). It will make local development more difficult, but I guess there's no better way.